domoticz.sh

```diff
@@ -15,14 +15,18 @@ PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin
 DESC="Domoticz Home Automation System"
 NAME=domoticz
 USERNAME=pi
-PIDFILE=/var/run/$NAME.pid
+# Use this PID file to run as regular user
+PIDFILE=/tmp/$NAME-$USERNAME.pid
+
 SCRIPTNAME=/etc/init.d/$NAME
+SSLCERT=/home/$USERNAME/domoticz/server_cert.pem
+
 
 DAEMON=/home/$USERNAME/domoticz/$NAME
 DAEMON_ARGS="-daemon"
-#DAEMON_ARGS="$DAEMON_ARGS -daemonname $NAME -pidfile $PIDFILE"
+DAEMON_ARGS="$DAEMON_ARGS -daemonname $NAME -pidfile $PIDFILE"
 DAEMON_ARGS="$DAEMON_ARGS -www 8080"
-DAEMON_ARGS="$DAEMON_ARGS -sslwww 443"
+DAEMON_ARGS="$DAEMON_ARGS -sslwww 10443 -sslcert $SSLCERT"
```

Comment:
This will break all current raspberry installations where people expect it to be found on port 443 (and probably other installations as well)
In the source it should load the ssl certification from the path where the executable resides (/home/$USERNAME/domoticz), this has always worked before...
need to double check this

Reply:
I realise this will break compatibility, so it shouldn't be default or at least give a warning.

Regarding server_cert.pem: I found it in the source here: (https://github.com/domoticz/domoticz/blob/72aea2965b0bf63b2b7be3fcbd8d9cb215bf68bf/main/mainworker.cpp#L211). It's a relative path and appears to depend on the location where domoticz is run: if I run it manually (/home/domoticz/domoticz/domoticz -daemonname domoticz -pidfile /tmp/domoticz-domoticz.pid -www 0 -sslwww 10443) it doesn't work if I'm not in the domoticz dir, but if I start it from /etc/init.d/domoticz.sh it does work. Perhaps it's an idea to add this to `/etc/domoticz/setupVars.conf` to make it an absolute path.

Reply:
Nope, best to use /etc/domoticz.conf, see the example configuration in the git repo.